google/cloud/spanner_v1/pool.py

```diff
@@ -515,7 +515,6 @@ def begin_pending_transactions(self):
         """Begin all transactions for sessions added to the pool."""
         while not self._pending_sessions.empty():
             session = self._pending_sessions.get()
-            session._transaction.begin()
```

Comment:
The deprecation warning looks good! The final question is whether we remove this begin or not.

If we keep it in then the pool's performance remains unchanged and we can have the deprecation notice recommend moving to `PingingPool`. This does mean there may be begin statements that aren't inlined.

If we remove it then we could potentially regress the pool's performance and in a way we would be forcibly migrating users to a different pool. But all the begin statements would be inlined.

@asthamohta As the current owner, do you have a preference here?

Reply:
@larkee I don't think performance will be impacted because we are not making any extra rpc call for begin transaction. Also for LAR feature , ILB needs to be implemented for all the pools.

Reply:
@larkee I feel it is okay to remove it as there shouldn't be any performance issues as surbhi has mentioned above.

Reply:
SGTM :+1: